src/mod_zone_difficulty_scripts.cpp

```diff
@@ -93,23 +152,469 @@ void ZoneDifficulty::LoadMapDifficultySettings()
+{
+    if (QueryResult result = CharacterDatabase.Query("SELECT * FROM zone_difficulty_hardmode_score"))
+    {
+        do
+        {
+            uint32 GUID = (*result)[0].Get<uint32>();
+            uint8 Type = (*result)[1].Get<uint8>();
+            uint32 Score = (*result)[2].Get<uint32>();
+
+            //LOG_INFO("module", "MOD-ZONE-DIFFICULTY: Loading from DB for player with GUID {}: Type = {}, Score = {}", GUID, Type, Score);
+            sZoneDifficulty->HardmodeScore[GUID][Type] = Score;
+
         } while (result->NextRow());
     }
 }
 
+/**
+ *  @brief Sends a whisper to all members of the player's raid in the same instance as the creature.
+ *
+ *  @param message The message which should be sent to the <Player>.
+ *  @param creature The creature who sends the whisper.
+ *  @param player The object of the player, whose whole group should receive the message.
+ */
+void ZoneDifficulty::SendWhisperToRaid(std::string message, Creature* creature, Player* player)
+{
+    Group::MemberSlotList const& members = player->GetGroup()->GetMemberSlots();
+    for (auto& member : members)
+    {
+        Player* mplr = ObjectAccessor::FindConnectedPlayer(member.guid);
+        if (creature && mplr && mplr->GetMap()->GetInstanceId() == player->GetMap()->GetInstanceId())
+        {
+            //LOG_INFO("module", "MOD-ZONE-DIFFICULTY: Player {} should receive a whisper.", mplr->GetName());
+            creature->Whisper(message, LANG_UNIVERSAL, mplr);
+        }
+    }
+}
+
+std::string ZoneDifficulty::GetItemTypeString(uint32 type)
+{
+    std::string typestring;
+    switch (type)
+    {
+    case ITEMTYPE_MISC:
+        typestring = "Back, Finger, Neck, and Trinket";
+        break;
+    case ITEMTYPE_CLOTH:
+        typestring = "Cloth";
+        break;
+    case ITEMTYPE_LEATHER:
+        typestring = "Leather";
+        break;
+    case ITEMTYPE_MAIL:
+        typestring = "Mail";
+        break;
+    case ITEMTYPE_PLATE:
+        typestring = "Plate";
+        break;
+    case ITEMTYPE_WEAPONS:
+        typestring = "Weapons and Shields";
+        break;
+    default:
+        LOG_ERROR("module", "MOD-ZONE-DIFFICULTY: Unknown type {} in ZoneDifficulty::GetItemTypeString.", type);
+    }
+    return typestring;
+}
+
+std::string ZoneDifficulty::GetContentTypeString(uint32 type)
+{
+    std::string typestring;
+    switch (type)
+    {
+    case TYPE_VANILLA:
+        typestring = "for Vanilla dungeons.";
+        break;
+    case TYPE_RAID_MC:
+        typestring = "for Molten Core.";
+        break;
+    case TYPE_RAID_ONY:
+        typestring = "for Onyxia.";
+        break;
+    case TYPE_RAID_BWL:
+        typestring = "for Blackwing Lair.";
+        break;
+    case TYPE_RAID_ZG:
+        typestring = "for Zul Gurub.";
+        break;
+    case TYPE_RAID_AQ20:
+        typestring = "for Ruins of Ahn'Qiraj.";
+        break;
+    case TYPE_RAID_AQ40:
+        typestring = "for Temple of Ahn'Qiraj.";
+        break;
```

Comment:
You can short  it  if the two cases are identical
```suggestion
    case TYPE_RAID_AQ20:
    case TYPE_RAID_AQ40:
        typestring = "for Ruins of Ahn'Qiraj.";
        break;
```

Reply:
They are different though ;)

Reply:
lol, sorry 😂